### PR TITLE
fix(api,api-client): Add environmentSlug in multiple places across the secret module

### LINKED ISSUE #486: API, API-CLIENT: Add `environmentSlug` in multiple places across the `secret` module
## Description
Add `slug` along with `environmentId` across multiple places in `secret` module. In simple terms, while returning the data, we are returning just the `environmentId`. We would instead like to return something like this:
```js
{
  //...
  environment: {
    id: '',
    slug: ''
  },
  //...
}
```

## Solution
List of files to update:
- https://github.com/keyshade-xyz/keyshade/blob/70ad4f7bd9637666e34878cb505782331d38798e/apps/api/src/secret/service/secret.service.ts#L127-L134
- https://github.com/keyshade-xyz/keyshade/blob/70ad4f7bd9637666e34878cb505782331d38798e/apps/api/src/secret/service/secret.service.ts#L253-L258
- https://github.com/keyshade-xyz/keyshade/blob/70ad4f7bd9637666e34878cb505782331d38798e/apps/api/src/secret/service/secret.service.ts#L696-L698
- https://github.com/keyshade-xyz/keyshade/blob/70ad4f7bd9637666e34878cb505782331d38798e/apps/api/src/secret/service/secret.service.ts#L503-L519
- Update the tests in secret.e2e.spec.ts to display these changes.
- Update the types in api-client package under secret.ts
- Update the tests in api-client package under secret.spec.ts

### PR DESCRIPTION
**Pull request by anudeeps352**
## Description

Add environmentSlug in multiple places across the secret module <br />

Fixes #486 

## Developer's checklist

- [X] My PR follows the style guidelines of this project
- [X] I have performed a self-check on my work

**If changes are made in the code:**

- [X] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [X] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works
- [ ] I have added relevant screenshots in my PR
- [X] There are no UI/UX issues

### Documentation Update

- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [ ] I have made the necessary updates to the documentation, or no documentation changes are required.

___

### **PR Type**
Enhancement, Tests

___

### **Description**
- Enhanced the secret module to include `environmentSlug` alongside `environmentId` in various places.
- Updated the secret service to select and return both `id` and `slug` for environments.
- Modified API client types to reflect the inclusion of `environment` details in secret versions.
- Added and updated tests in both the API and API client to verify the presence of `environment.id` and `environment.slug`.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>secret.e2e.spec.ts</strong><dd><code>Add tests for environment slug and id in secret versions</code>&nbsp; </dd></summary>
<hr>

apps/api/src/secret/secret.e2e.spec.ts

<li>Added tests to verify <code>environment.id</code> and <code>environment.slug</code> in secret <br>versions.<br> <li> Included <code>environment</code> in the Prisma query for secret versions.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/509/files#diff-598db59202ce7ecc8d2fb84528577afd14eac8275a30e174d704d6fc2b7f5c3b">+10/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>secret.spec.ts</strong><dd><code>Add tests for environment slug and id in API client</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/api-client/tests/secret.spec.ts

<li>Added tests to verify <code>environment.id</code> and <code>environment.slug</code> in API <br>client.<br> <li> Initialized <code>environment</code> variable for test setup.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/509/files#diff-4f38527eb4c63b4ddb03d7c6e668cc4bce26acafa0352464405521f47d7868f5">+5/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>secret.service.ts</strong><dd><code>Include environment slug and id in secret service</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/api/src/secret/service/secret.service.ts

<li>Updated secret service to include <code>environment.id</code> and <code>environment.slug</code>.<br> <li> Modified Prisma queries to select environment details.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/509/files#diff-dee38177617754972e3d3f727d7f1536566d7a784b7ffdda74aa97d6eec4cbc5">+39/-3</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>secret.types.d.ts</strong><dd><code>Update secret types to include environment slug</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/api-client/src/types/secret.types.d.ts

<li>Updated <code>Secret</code> and <code>UpdateSecretResponse</code> interfaces to include <br><code>environment</code> object.<br> <li> Added <code>slug</code> to the <code>environment</code> object in types.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/509/files#diff-d0148d6535d1f0c29d4022e1be61df8ea298e755b37975b0238a0070510b6dc2">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: apps/api/src/secret/service/secret.service.ts [hunk 1]
```diff
@@ -648,6 +666,16 @@ export class SecretService {
             id: true,
             name: true
           }
+        },
+        versions: {
+          select: {
+            environment: {
+              select: {
+                id: true,
+                slug: true
```
Comment (rajdip-b) [inline apps/api/src/secret/service/secret.service.ts:675]: You would also need to include the `name` parameter in here.

File: packages/api-client/tests/secret.spec.ts [hunk 2]
```diff
@@ -12,6 +12,7 @@ describe('Secret Controller Tests', () => {
   let workspaceSlug: string | null
   let environmentSlug: string | null
   let secretSlug: string | null
+  let environment
```
Comment (rajdip-b) [inline packages/api-client/tests/secret.spec.ts:15]: add type

File: packages/api-client/tests/secret.spec.ts [hunk 3]
```diff
@@ -58,7 +59,8 @@ describe('Secret Controller Tests', () => {
       )
     ).json()) as any
 
-    environmentSlug = createEnvironmentResponse.slug
+    environment = createEnvironmentResponse
+    environmentSlug = environment.slug
```
Comment (rajdip-b) [inline packages/api-client/tests/secret.spec.ts:63]: Maybe we can get rid of `environmentSlug` and keep just `environment: Environment`

**Review by rajdip-b — CHANGES_REQUESTED** [review]

File: packages/api-client/tests/secret.spec.ts (hunk 3 above)
Comment (anudeeps352) [reply packages/api-client/tests/secret.spec.ts:63]: so remove environmentSlug and just use environment.slug in related segements?

File: packages/api-client/tests/secret.spec.ts (hunk 2 above)
Comment (anudeeps352) [reply packages/api-client/tests/secret.spec.ts:15]: or is just type:any fine ?? cause in other files its so

Comment (anudeeps352) [reply packages/api-client/tests/secret.spec.ts:15]: Ig environment: Environment ,imported from '@api-client/types/environment.types' works

File: packages/api-client/tests/secret.spec.ts (hunk 3 above)
Comment (rajdip-b) [reply packages/api-client/tests/secret.spec.ts:63]: yes

File: packages/api-client/tests/secret.spec.ts (hunk 2 above)
Comment (rajdip-b) [reply packages/api-client/tests/secret.spec.ts:15]: you are right, that almost left my mind. do one thing, don't compare the ID, just compare the slugs. and also, just store the `environmentSlug` in here

Comment (rajdip-b) [reply packages/api-client/tests/secret.spec.ts:15]: You dont have to store the environment id. we are already storing the environment slug so you can just do that comparison and drop the id comparison.

File: packages/api-client/tests/secret.spec.ts [hunk 4]
```diff
@@ -10,9 +11,8 @@ describe('Secret Controller Tests', () => {
   const email = 'johndoe@example.com'
   let projectSlug: string | null
   let workspaceSlug: string | null
-  let environmentSlug: string | null
   let secretSlug: string | null
-  let environment
+  let environment: Environment
```
Comment (rajdip-b) [inline packages/api-client/tests/secret.spec.ts:15]: Remove this line, and keep `environmentSlug`. Make all the comparisons on `environmentSlug`. We don't want to compare ID anymore

**Review by rajdip-b — CHANGES_REQUESTED** [review]

**Review by rajdip-b — APPROVED** [review]
LGTM

**Pull request merged**